packages/keystone/apolloErrorFormatter.js

```diff
@@ -96,130 +97,222 @@ function _patchKnownErrorCases (error, result) {
     }
 }
 
-function _extractInnerGQLError (error) {
-    const innerErrors = get(error, 'errors', [])
-    return innerErrors.find(err => get(err, 'name') === 'GQLError')
+class NonError extends Error {
+    constructor (message) {
+        super(util.inspect(message))
+        this.name = 'NonError'
+        Error.captureStackTrace(this, NonError)
+    }
 }
 
-/**
- * Use it if you need to safely prepare error for logging or ApolloServer result
- * @param {Error} error -- any error
- * @param {Boolean} hideInternals -- do you need to hide some internal error fields
- * @param {Boolean} applyPatches -- do you need to apply a common error message patches
- * @returns {import('graphql').GraphQLFormattedError}
- */
-const safeFormatError = (error, hideInternals = false, applyPatches = true) => {
-    const result = {}
+function _ensureError (error) {
+    if (!(error instanceof Error)) {
+        return new NonError(error)
+    }
+    return error
+}
 
-    // error keyst: message, name, stack
-    const pickKeys1 = (hideInternals) ? ['message', 'name'] : ['message', 'name', 'stack']
-    Object.assign(result, pick(ensureError(error), pickKeys1))
+function _buildCombinedStacks (stack, errors) {
+    if (!errors || !isArray(errors)) return stack
+    const result = (stack) ? [stack] : []
```

Comment:
Same about JSDoc, if stack is Array as many can suppose, why its wrapped in another array?